juju/model.py

```diff
@@ -772,9 +772,13 @@ async def watch_received_waiter():
             raise self._watcher_task.exception()
 
         if self.info is None:
```

Comment:
It seems odd to me that we check "self.info is None" but we then set "self._info". I realize that info is a property that just returns self._info, but it seems much better to check the same variable that you are setting. (it becomes much clearer that you are using a cache variable)

Otherwise I would put the caching into the info property itself, so that only when we need self.info do we write to self._info.
And then this code becomes:
```
if self.uuid is None:
  self.uuid = self.info.uuid
```

and the property becomes:
```
@property
def info(self):
    if self._info is None:
            async with ConnectedController(self.connection()) as contr:
                self._info = await contr.get_model_info(model_name, model_uuid)
                log.debug('Got ModelInfo: %s', vars(self.info))
   return self._info
```

Then we cache whenever anyone accesses self.info, but not until then, and we don't need it if we already know the uuid.

Thoughts?

Reply:
Yeah I agree, but I'd probably go with just checking the `self._info`, because caching can't be done in the property method (you can't do async stuff in a non-async method).